exonum-java-binding/core/src/main/java/com/exonum/binding/core/transaction/TransactionContext.java

```diff
@@ -50,6 +50,18 @@
    */
   PublicKey getAuthorPk();
 
+  /**
+   * Returns the name of the service instance.
+   */
+  String getServiceName();
+
+  /**
+   * Returns the numeric id of the service instance.
+   *
+   * @see TransactionMessage#getServiceId()
+   */
+  Integer getServiceId();
```

Comment:
can it be `null`? If not why doesn't `int`?

Reply:
I intentionally used `Integer` here, so that if `serviceId` isn't specified, NPE would be thrown by AutoValue when creating an instance. Opposed to `int`, which would just set this value to `0` by default, which is error-prone. Should I keep the `Integer` or change it to `int` and validate that it was set or change it to `int` and not bother with that?

Reply:
But the builder checks (or shall check) that, doesn't it?

Reply:
The builder implementation nuances (that it uses Integer to distinguish between a set and unset value) must not affect the API.